src/main.rs

```diff
@@ -139,12 +139,32 @@ async fn insert_receipts(
     chunks: &[near_indexer::IndexerChunkView],
     local_receipts: &[near_indexer::near_primitives::views::ReceiptView],
     pool: &Pool<ConnectionManager<PgConnection>>,
+    outcomes: &HashMap<
+        near_indexer::near_primitives::hash::CryptoHash,
+        near_indexer::near_primitives::views::ExecutionOutcomeWithIdView,
+    >,
 ) {
+    fn receipt_status(
+        outcomes: &HashMap<
+            near_indexer::near_primitives::hash::CryptoHash,
+            near_indexer::near_primitives::views::ExecutionOutcomeWithIdView,
+        >,
```

Comment:
Let's define a meaningful type-alias:

```rust
/// Map Receipt ID to Execution Outcome
pub type ExecutionOutcomesByReceiptId = HashMap<
            near_indexer::near_primitives::hash::CryptoHash,
            near_indexer::near_primitives::views::ExecutionOutcomeWithIdView,
        >;
```

I would even argue to define it on the streamer/indexer-framework side.